src/librustdoc/clean/mod.rs

```diff
@@ -2313,23 +2313,30 @@ impl Clean<Item> for (&hir::ForeignItem<'_>, Option<Ident>) {
     }
 }
 
-impl Clean<Item> for doctree::Macro {
+impl Clean<Item> for (&hir::MacroDef<'_>, Option<Ident>) {
     fn clean(&self, cx: &DocContext<'_>) -> Item {
+        let (item, renamed) = self;
+        let name = renamed.unwrap_or(item.ident).name;
+        let tts = item.ast.body.inner_tokens().trees().collect::<Vec<_>>();
+        // Extract the spans of all matchers. They represent the "interface" of the macro.
+        let matchers = tts.chunks(4).map(|arm| arm[0].span()).collect::<Vec<_>>();
+
         Item::from_def_id_and_parts(
-            self.def_id,
-            Some(self.name.clean(cx)),
+            cx.tcx.hir().local_def_id(item.hir_id).to_def_id(),
+            Some(name.clean(cx)),
             MacroItem(Macro {
+                // FIXME(#76761): Make this respect `macro_rules!` vs `pub macro`
                 source: format!(
                     "macro_rules! {} {{\n{}}}",
```

Comment:
If you want, I think this would be pretty easy to add now by looking at `ast.macro_rules`: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast/ast/struct.MacroDef.html. Feel free to make that change in a later PR if you don't feel confident changing multiple things at once, though.

Reply:
I'm fine adding it now, just wasn't sure if you minded the scope